### PR TITLE
update #993 - fix rendering bug on challenge page

### PR DESCRIPTION
**Pull request by Ulisseus**
What happened? I accidentally passed initial value instead of state to query, so when user clicked on different challenge diff render wasn't updated (props were the same). And query failed (one of arguments was undefined).  

Also, with recent dayjs update we got a bunch of failing snapshots. I could mock date but I use it as unique id in one of test cases to break user's cache. I leave it as it is for now.

### CODE REVIEW

File: components/__snapshots__/ChallengeMaterial.test.js.snap [hunk 1]
```diff
@@ -86,26 +86,12 @@ exports[`Curriculum challenge page Should be able to select another challenge 1`
             <p
               class="selection__title text-muted mx-2"
             >
-              Select submission
+              Loading previous submissions...
```
Comment (ggwadera) [inline components/__snapshots__/ChallengeMaterial.test.js.snap:89]: Looks like this snapshot is incorrect.

Comment (Ulisseus) [reply components/__snapshots__/ChallengeMaterial.test.js.snap]: Fixed.

**Comment by Ulisseus** [comment]
I changed dummy data to avoid shifting dates in snapshots. Ideally dates should be mocked with `jest.useFakeTimers('modern')` but using fake timers will cause problems with breaking cache.

**Comment by ggwadera** [comment]
Why `jest.useFakeTimers`? That's used for testing code with timed functions like `setTimeout` and not dates. You'll probably need to mock `dayjs` to have some controlled way of outputting dates or try to change the tests to not use snapshots.

**Review by ggwadera — APPROVED** [review]

**Comment by Ulisseus** [comment]
>You'll probably need to mock dayjs to have some controlled way of outputting dates or try to change the tests to not use snapshots.
I tried it but mocking dayjs is not as easy as it sounds (days js uses another library for mocking in its own tests). 

I updated this pr again, so dummy data uses fixed dates `createdAt: dayjs().subtract(16, 'day').valueOf()`. It's 16 days ago no matter what.

**Pull request merged**